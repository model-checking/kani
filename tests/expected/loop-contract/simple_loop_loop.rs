// Copyright Kani Contributors
// SPDX-License-Identifier: Apache-2.0 OR MIT

// kani-flags: -Z loop-contracts

//! Check if loop-loop invariant is correctly applied.

#![feature(proc_macro_hygiene)]
#![feature(stmt_expr_attributes)]

#[kani::proof]
fn main() {
    let mut i = 100;
    #[kani::loop_invariant(i<=100 && i >= 4 && i % 2 ==0)]
    loop {
        i = i - 2;
        if i == 2 {
            break;
        }
    }
    assert!(i == 2);
}
