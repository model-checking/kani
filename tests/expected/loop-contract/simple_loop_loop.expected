main.loop_invariant_base.1\
	 - Status: SUCCESS\
	 - Description: "Check invariant before entry for loop main.0"\
in function main

VERIFICATION:- SUCCESSFUL
